### PR TITLE
chore: Sends Slack notifications when PR is ready for review

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Sends Slack notifications when PR is ready for review.

`skip-slack-notification` label can be used when no automatic notifications want to be sent.

This PR is a follow-up to https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2168

Link to any related issue(s): CLOUDP-245726

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

**Review by EspenAlbert — APPROVED** [review]
Nice 👍 
Later, we could have a follow-up PR to disable the message if a label exists on the PR.

E.g., imagine you are doing multiple small PRs and want to send a shared message instead of 1 message per PR

**Comment by lantoli** [comment]
@EspenAlbert it's already done, label skip-slack-notification disables it

File: .github/workflows/slack-notification.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
+name: Slack Notification
+
+on:
+  pull_request:
+    types: [opened, ready_for_review]
+
+jobs:
+  precheck:
+    if: ${{ github.event.pull_request.draft == false && !contains(github.event.pull_request.labels.*.name, 'skip-slack-notification') }}
```
Comment (lantoli) [inline .github/workflows/slack-notification.yml:9]: @EspenAlbert FYI label check

Comment (EspenAlbert) [reply .github/workflows/slack-notification.yml:9]: Ahh. Nice 👏

**Review by gssbzn — COMMENTED** [review]
why not simply use `/github subscribe`?

**Comment by lantoli** [comment]
@gssbzn I'm not sure we could get the same level of low noise

**Comment by gssbzn** [comment]
>I'm not sure we could get the same level of low noise

have you tried?

**Pull request closed without merging**